satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -0,0 +1,547 @@
+            a, b = self.nc['a_wv'], self.nc['b_wv']
+        else:
+            a, b = self.nc['a_ir'], self.nc['b_ir']
+        rad = a + b * counts
+        return rad.where(rad > 0)
+
+    def _ir_wv_radiance_to_brightness_temperature(self, rad, channel):
+        """Convert IR/WV radiance to brightness temperature.
+
+        Reference: [PUG], equations (5.1) and (5.2).
+        """
+        if channel == 'WV':
+            a, b = self.nc['bt_a_wv'], self.nc['bt_b_wv']
+        else:
+            a, b = self.nc['bt_a_ir'], self.nc['bt_b_ir']
+        bt = b / (np.log(rad) - a)
+        return bt.where(bt > 0)
+
+    def _mask(self, ds):
+        """Mask pixels with bad quality.
+
+        Pixels are considered bad quality if the "quality_pixel_bitmask" is
+        everything else than 0 (no flag set) or 2 ("use_with_caution" and no
+        other flag set).
+        """
+        mask = self._get_mask(ds)
+        return ds.where(np.logical_or(mask == 0, mask == 2))
+
+    @shape_cache
+    def _get_mask(self, ds):
+        """Get quality bitmask for the given dataset."""
+        mask = self.nc['quality_pixel_bitmask']
+
+        # Mask has high (VIS) resolution. Downsample to low (IR/WV) resolution
+        # if required.
+        if mask.coords['y'].size > ds.coords['y'].size:
+            mask = mask.isel(y=slice(None, None, 2),
+                             x=slice(None, None, 2))
+            mask = mask.assign_coords(y=ds.coords['y'].values,
+                                      x=ds.coords['x'].values)
+        return mask
+
+    @shape_cache
+    def _get_acq_time(self, ds):
+        """Get scanline acquisition time.
+
+        The files provide timestamps per pixel for the low resolution
+        channels (IR/WV) only.
+
+        1) Average values in each line to obtain one timestamp per line.
+        2) For the VIS channel duplicate values in y-direction (as
+           advised by [PUG]).
+
+        Note that the timestamps do not increase monotonically with the
+        line number in some cases.
+
+        Returns:
+            Mean scanline acquisition timestamps
+        """
+        # Variable is sometimes named "time" and sometimes "time_ir_wv".
+        try:
+            time_lores = self.nc['time_ir_wv']
+        except KeyError:
+            time_lores = self.nc['time']
+
+        # Compute mean timestamp per scanline
+        time_lores = time_lores.mean(dim='x_ir_wv').rename({'y_ir_wv': 'y'})
+
+        # If required, repeat timestamps in y-direction to obtain higher
+        # resolution
+        y_lores = time_lores.coords['y'].values
+        y_target = ds.coords['y'].values
+        if y_lores.size < y_target.size:
+            reps = y_target.size // y_lores.size
+            y_lores_rep = np.repeat(y_lores, reps)
+            time_hires = time_lores.reindex(y=y_lores_rep)
+            time_hires = time_hires.assign_coords(y=y_target)
+            return time_hires
+
+        return time_lores
```

Comment:
This function could be refactored into getting the low resolution time, and upsampling.

Reply:
Done!